src/extension/monitor/hibernate/src/main/java/org/geoserver/monitor/hib/MonitoringDataSource.java

```diff
@@ -108,20 +107,20 @@ void initializeDataSource() throws Exception {
         }
     }
     
-    void configureDataSource(File dbprops, File monitoringDir) throws Exception {
+    void configureDataSource(Resource dbprops, Resource monitoringDir) throws Exception {
         Properties db = new Properties();
 
         if (dbprops == null) {
-            dbprops = new File(monitoringDir, "db.properties");
+            dbprops = monitoringDir.get("db.properties");
             
             //use a default, and copy the template over
             InputStream in = getClass().getResourceAsStream("db.properties");
-            IOUtils.copy(in, new FileOutputStream(dbprops));
+            IOUtils.copy(in, dbprops.out());
             
             db.load(getClass().getResourceAsStream("db.properties"));
         }
         else {
-            FileInputStream in = new FileInputStream(dbprops);
+            InputStream in = dbprops.in();
             db.load(in);
             in.close();
```

Comment:
@NielsCharlier is this a resource leak? Could it be fixed by a try-with-resources block?